File: assignment-3/date-servers-part-2/python/Datatypes/HttpResponse.py
from pydantic import BaseModel


class HttpResponse():
    status: int
    message: str

    def __init__(self, status, message):
        self.status = status
        self.message = message

    def __repr__(self) -> str:
        return f'"status": {self.status}, "message": {self.message}'

    def __iter__(self):
        yield 'status', self.status
        yield 'message', self.message

    def internal_server_error(self):
        self.status = 500
        self.message = "Internal Server Error"


class HttpResponseExample(BaseModel):
    status: int = 500
    message: str = "Internal Server Error"

File: assignment-3/date-servers-part-2/rust/src/data_types/http_response.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Serialize, Deserialize, Debug, ToSchema)]
pub struct CustomHttpResponse {
    pub status: u16,
    pub message: String,
}

#[allow(dead_code)]
impl CustomHttpResponse {
    pub fn new<T: ToString>(code: u16, message: T) -> Self {
        CustomHttpResponse {
            status: code,
            message: message.to_string(),
        }
    }

    pub fn not_found() -> Self {
        CustomHttpResponse {
            status: 404,
            message: String::from("Not found."),
        }
    }

    pub fn internal_server_error() -> Self {
        CustomHttpResponse {
            status: 500,
            message: String::from("Internal Server Error."),
        }
    }
}
